tests/test_bar_data.py

```diff
@@ -0,0 +1,429 @@
+            cls.env,
+            equity_daily_reader=cls.build_daily_data()
+        )
+
+        cls.ASSET1 = cls.env.asset_finder.retrieve_asset(1)
+        cls.ASSET2 = cls.env.asset_finder.retrieve_asset(2)
+
+        cls.ASSETS = [cls.ASSET1, cls.ASSET2]
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.tempdir.cleanup()
+
+    @classmethod
+    def build_daily_data(cls):
+        path = cls.tempdir.getpath("testdaily.bcolz")
+
+        dfs = {
+            1: create_daily_df_for_asset(cls.env, cls.days[0], cls.days[-2]),
+            2: create_daily_df_for_asset(
+                cls.env, cls.days[0], cls.days[-2], interval=2
+            )
+        }
+
+        daily_writer = DailyBarWriterFromDataFrames(dfs)
+        daily_writer.write(path, cls.days[:-1], dfs)
+
+        return BcolzDailyBarReader(path)
+
+    def test_day_before_assets_trading(self):
+        # use the day before self.days[0]
+        day = self.env.previous_trading_day(self.days[0])
+
+        bar_data = BarData(self.data_portal, lambda: day, "daily")
+        self.check_internal_consistency(bar_data)
+
+        self.assertFalse(bar_data.can_trade(self.ASSET1))
+        self.assertFalse(bar_data.can_trade(self.ASSET2))
+
+        self.assertFalse(bar_data.is_stale(self.ASSET1))
+        self.assertFalse(bar_data.is_stale(self.ASSET2))
+
+        for field in ALL_FIELDS:
+            for asset in self.ASSETS:
+                asset_value = bar_data.spot_value(asset, field)
+
+                if field in OHLCP:
+                    self.assertTrue(np.isnan(asset_value))
+                elif field == "volume":
+                    self.assertEqual(0, asset_value)
+                elif field == "last_traded":
+                    self.assertTrue(asset_value is pd.NaT)
+
+    def test_semi_active_day(self):
+        # on self.days[0], only asset1 has data
+        bar_data = BarData(self.data_portal, lambda: self.days[0], "daily")
+        self.check_internal_consistency(bar_data)
+
+        self.assertTrue(bar_data.can_trade(self.ASSET1))
+        self.assertFalse(bar_data.can_trade(self.ASSET2))
+
+        # because there is real data
+        self.assertFalse(bar_data.is_stale(self.ASSET1))
+
+        # because there has never been a trade bar yet
+        self.assertFalse(bar_data.is_stale(self.ASSET2))
+
+        self.assertEqual(3, bar_data.spot_value(self.ASSET1, "open"))
+        self.assertEqual(4, bar_data.spot_value(self.ASSET1, "high"))
+        self.assertEqual(1, bar_data.spot_value(self.ASSET1, "low"))
+        self.assertEqual(2, bar_data.spot_value(self.ASSET1, "close"))
+        self.assertEqual(200, bar_data.spot_value(self.ASSET1, "volume"))
+        self.assertEqual(2, bar_data.spot_value(self.ASSET1, "price"))
+        self.assertEqual(self.days[0],
+                         bar_data.spot_value(self.ASSET1, "last_traded"))
+
+        for field in OHLCP:
+            self.assertTrue(np.isnan(bar_data.spot_value(self.ASSET2, field)))
```

Comment:
So that it can be seen which if any values fail. It would be helpful here to either include the field as the 3rd argument to the assert, or `np.testing.assert_array_equal` does support checking `np.nan`s against eachother, so this could be.

```
np.testing.assert_array_equal(
   [np.nan] * 5,
   [bar_data.spot_value(self.ASSET2, field) for field in OHLCP]
)
```

 

```